### DISCUSSION TITLE
alpine laravel multiple checkbox duplicate values

### DISCUSSION THREAD

**Discussion by peecota** [alpinejs/alpine#3931] [1. Help] [answered]
"Indeed, I aim to select multiple items from a datatable constructed with Livewire 3 using Alpine.js. The following code establishes an array named 'selection.' Using x-model, I populate the array when checking a checkbox in a row. However, the issue arises when navigating to other pages; the code begins to introduce duplicate values into the array, as depicted in the attached image. Here is my code." 
ps: this problem doesnt occur where i remove the id of the checkbox. but when i click on a checkbox., the similar rows of the other pages  are checked at the sametime

for the alpine version <script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js"></script>

<div class="col-lg-9 col-md-12 col-12"
        x-data = "{selection:[]}">
    <div class="main-content">
        <div class="dashboard-block mt-0">
            
            <div class="my-items">
                
                <table class="table">
                    <thead>
                      <tr>
                        <th scope="col"><input type="checkbox" class="form-check-input" id="selectAllCheckbox"></th>
                        <th scope="col">title</th>
                        <th scope="col">price</th>
                        <th scope="col">category</th>
                      </tr>
                    </thead>
                    @foreach ($this->listings as $listing)
                    <tbody>
                      <tr>
                        <th scope="row"><input type="checkbox" name="ids" class="form-check-input" 
                            id="{{$listing->id}}" value="{{$listing->id}}" x-model.fill="selection"></th>
                        <td>{{$listing->title}}</td>
                        <td>{{$listing->price}}</td>
                        <td>{{$listing->category->name}}</td>
                      </tr>
                    </tbody>
                    @endforeach
                  </table>

                
                <div class="pagination left">
                    {{ $this->listings->links() }}
                </div>

                
            </div>
            
        </div>
    </div>
</div>

![image](https://github.com/alpinejs/alpine/assets/43576819/6ddb7b27-cc92-40ca-92b5-71aa85633ab3)

**Comment by joshhanley** [comment, marked as answer]
@peecota your code isn't showing up  properly in your description above.

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

In regards to your issue, it sounds like a morph problem. You need to ensure you have `wire:key` on elements in a loop. See here for details https://livewire.laravel.com/docs/components#adding-wirekey-to-foreach-loops

You also mention you are using Alpine from a CDN, but Alpine V3 is bundled with Livewire V3, so including Alpine via a CDN will also be causing issues. So you should remove that.

Also things like this should probably be on the Livewire repo in future.

Hope this helps!

**Comment by peecota** [comment]
thank you a lot, the problem was related to the CDN.  and it was resolved by deleting it
